src/Integration/RequestFactory.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Integration;
+
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * Allows customizing the request information that is attached to the logged event.
+ * An implementation of this interface can be passed to RequestIntegration.
+ */
+interface RequestFactory
```

Comment:
What about renaming this interface to `RequestFetcherInterface`? A factory in this case seems to imply that we are really creating a request while probably we just want to fetch it from some other place